File: util/src/test/java/dev/openrs2/util/StringUtilsTest.java
package dev.openrs2.util;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

public final class StringUtilsTest {
	@Test
	public void testIndefiniteArticle() {
		assertEquals("a", StringUtils.indefiniteArticle("book"));
		assertEquals("an", StringUtils.indefiniteArticle("aeroplane"));

		assertThrows(IllegalArgumentException.class, () -> StringUtils.indefiniteArticle(""));
	}

	@Test
	public void testCapitalize() {
		assertEquals("Hello", StringUtils.capitalize("hello"));
		assertEquals("", StringUtils.capitalize(""));
	}
}

File: util/src/test/java/dev/openrs2/util/StringUtilsTest.kt
package dev.openrs2.util

import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFailsWith

object StringUtilsTest {
    @Test
    fun testIndefiniteArticle() {
        assertEquals("a", "book".indefiniteArticle())
        assertEquals("an", "aeroplane".indefiniteArticle())
        assertFailsWith(IllegalArgumentException::class) {
            "".indefiniteArticle()
        }
    }
}
